contributors/devel/kubectl-conventions.md

```diff
@@ -375,6 +376,21 @@ this is not a mandatory structure and not every command is implemented this way,
 but this is a nice convention so try to be compliant with it. As an example,
 have a look at how [kubectl logs](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/logs.go) is implemented.
 
+## Exit code conventions
+
+Generally, for all the commands exit code, result of `zero` means succeed and `non-zero` means error.
+
+For some declarative commands like `apply` may have a situation which command exited success but without actual changes, we provide a flag `--error-unchanged` to specify the exit code to 2 if necessary.
```

Comment:
For idempotent operations, requiring no changes *is* success (apply is a "make it so" command… if it is already so, a non-zero exit code makes sense)

Reply:
Yeah, please rephrase as Jordan mentioned.  Idempotent ("make-it-so") commands should return 0 when successful (even if no changes were provided).  For commands that are not explicitly expected to be idempotent, consumers can request treating "already-so" as "make-it-so" via flag.

Reply:
combine two comments above then add a little redundancy and parallelism, ptal